src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -222,7 +222,7 @@ String MergeTreePartition::getID(const Block & partition_key_sample) const
     bool are_all_integral = true;
     for (const Field & field : value)
     {
-        if (field.getType() != Field::Types::UInt64 && field.getType() != Field::Types::Int64 && field.getType() != Field::Types::IPv4)
+        if (field.getType() != Field::Types::Bool && field.getType() != Field::Types::UInt64 && field.getType() != Field::Types::Int64 && field.getType() != Field::Types::IPv4)
```

Comment:
Why do you need this change?

If `read_bool_field_as_int = true` then bool will be deserialized using UInt8 serialization which will return UInt64 type and this should work?

Reply:
It's not needed indeed, I forgot to remove this change after testing. Thanks